components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1435,8 +1435,25 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
         const projects = await this.projectsService.getProjectsByCloneUrls(repositories.map(r => r.cloneUrl));
 
-        const cloneUrlsInUse = new Set(projects.map(p => p.cloneUrl));
-        repositories.forEach(r => { r.inUse = cloneUrlsInUse.has(r.cloneUrl) });
+        const cloneUrlToProject = new Map(projects.map(p => [p.cloneUrl, p]));
+
+        for (const repo of repositories) {
+            const p = cloneUrlToProject.get(repo.cloneUrl);
+            if (p) {
+                if (p.userId) {
+                    const owner = await this.userDB.findUserById(p.userId);
+                    if (owner) {
+                        repo.inUse = {
+                            userName: owner?.name || owner?.fullName || 'somebody'
+                        }
+                    }
+                } else if (p.owners && p.owners[0]) {
+                    repo.inUse = {
+                        userName: p.owners[0] || 'somebody'
+                    }
+                }
```

Comment:
I love that we now have a list of `owners` (or, more precisely, `teamOwners`) per project here.

However, it still seems very weird to randomly pick the first team owner and show it here:
- What if the first team owner has nothing to do with the project?
- What if the relevant person to contact is actually the 2nd or 3rd owner?
- What if the person who originally added the project since left the team?

Shouldn't we instead show the actual team name?
- That seems like the single most relevant information here: This project is already part of team A. If you want access to it, join team A
- We can still mention a random team owner (or all team owners) in a helpful hint e.g. "You can request a team invite from X, Y or Z")

Reply:
With that, you'd still not know who to contact. Ideally, we should have a [request access] button, that sends an email to the admins of the team to approve. Or even have people with a certain email domain auto-join.
But for the time beeing I think just showing a concrete name of someone who can add you seems like a good improvement over the current situation.

Reply:
cf. https://github.com/gitpod-io/gitpod/issues/7007

cc. @jankeromnes as he mentioned the same on slack

Reply:
To help with some of the questions from @jankeromnes in https://github.com/gitpod-io/gitpod/pull/5128/files#r767591699:

> What if the first team owner has nothing to do with the project?

This still provides some insight on who to reach, no?

> What if the relevant person to contact is actually the 2nd or 3rd owner?

Eventually, we'll allow users to requests access to teams as mentioned in https://github.com/gitpod-io/gitpod/pull/5128/files#r767696965 and https://github.com/gitpod-io/gitpod/pull/5128/files#r767698735.

> What if the person who originally added the project since left the team?

This is something that ideally we could reconfigure in project settings. Added https://github.com/gitpod-io/gitpod/issues/7220 to tack this. See also https://github.com/gitpod-io/gitpod/issues/7010.

🍊 🍊 🍊 🍊 

One issue with :a: requesting access to a team and :b: showing the team name instead is that users could add projects under their personal account which makes access requests impossible. One solution to this was recently discussed in a sync meeting and includes allowing _**only**_ projects from GitHub Organizations[[1](https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/about-organizations)] or GitLab Groups[[2](https://docs.gitlab.com/ee/user/group/)] to be part of teams. 💭  Cc @jldec